threescale_api/resources.py

```diff
@@ -509,6 +509,34 @@ def __init__(self, *args, entity_name='backend_api',
     def url(self) -> str:
         return self.threescale_client.admin_api_url + '/backend_apis'
 
+    def list(self, **kwargs):
+        return list(super().list(**kwargs))
+
+    def _list(self, **kwargs):
+        if "page" in kwargs.get("params", {}):
+            return super()._list(**kwargs)
+
+        pagenum = 1
+
+        kwargs = kwargs.copy()
+        if "params" not in kwargs:
+            kwargs["params"] = {}
+
+        kwargs["params"]["page"] = pagenum
+        kwargs["params"]["per_page"] = 500
+
+        page = super()._list(**kwargs)
+
+        while len(page):
+            for i in page:
+                yield i
+            pagenum += 1
+            kwargs["params"]["page"] = pagenum
+            page = super()._list(**kwargs)
```

Comment:
I would extract the pagination logic logic to a separate method

Reply:
It doesn'ŧ have to be easy to move it (at least I can't imagine easy way). Even current "iterator" was second attempt and that was because of `**kwargs` in `list()` `_list()` etc. My initial attempt to implement `__iter__()` broke that `**kwargs` passing. Also having `_list()` returning the interator (yeah, I know it's generator) is handy because of calls like `item in _list()`
Hint?

Reply:
hmm, ok you are right the "extraction" would not be as simple as I thought. Ok so we can go with this version for now